src/engine/guidance/post_processing.cpp

```diff
@@ -498,6 +498,10 @@ void collapseTurnAt(std::vector<RouteStep> &steps,
     BOOST_ASSERT(one_back_index < steps.size());
     const auto &current_step = steps[step_index];
     const auto &one_back_step = steps[one_back_index];
+    // Don't collapse roundabouts
+    if (entersRoundabout(current_step.maneuver.instruction) ||
+        entersRoundabout(one_back_step.maneuver.instruction))
+        return;
```

Comment:
Hmm do you know where we introduced the regression and why our tests did not catch it? Also why only the roundabout enter instruction - are the silent roundabout instructions already collapsed into the multi-hop roundabout instruction here?